catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java

```diff
@@ -0,0 +1,215 @@
+      Pattern.compile(
+          String.format(
+              "(%s)|(%s)", PATTERN_UTM_COORDINATE.pattern(), PATTERN_UPS_COORDINATE.pattern()),
+          Pattern.CASE_INSENSITIVE);
+
+  // This key tells the UI that the geo is on the suggestion itself
+  private static final String LITERAL_SUGGESTION_ID = "LITERAL-UTM-UPS";
+
+  private final CoordinateSystemTranslator translator;
+
+  public UtmUpsCoordinateProcessor(CoordinateSystemTranslator translator) {
+    this.translator = translator;
+  }
+
+  /**
+   * Given a list of {@link Suggestion}s and the query that yielded them, enhance the list with
+   * additional suggestions based upon the presence of coordinate literals in the query string.
+   *
+   * @param results the list of results to enhance.
+   * @param query the query string to search for coordinate literals.
+   */
+  public void enhanceResults(List<Suggestion> results, String query) {
+    LOGGER.trace("(UTM/UPS) Adding result for query [{}]", query);
+    final LiteralSuggestion literal = getUtmUpsSuggestion(query);
+    if (literal != null && literal.hasGeo()) {
+      LOGGER.trace("Adding the UTM/UPS suggestion to results [{}]", literal);
+      results.add(0, literal);
+    }
+    LOGGER.trace("(UTM/UPS) Done");
+  }
+
+  @Nullable
+  private LiteralSuggestion getUtmUpsSuggestion(String query) {
+    final Matcher matcher = PATTERN_UTM_OR_UPS_COORDINATE.matcher(query);
+    final StringBuilder nameBuilder = new StringBuilder("UTM/UPS:");
+    final List<UtmUpsCoordinate> utmUpsCoords = new ArrayList<>();
+    while (matcher.find()) {
+      final String group = matcher.group();
+      LOGGER.trace("Match found [{}]", group);
+      final String utmOrUpsText = normalizeCoordinate(group);
+      final UtmUpsCoordinate utmUps = parseUtmUpsString(utmOrUpsText);
+      if (utmUps != null) {
+        nameBuilder.append("[ ").append(utmUps.toString()).append(" ]");
+        utmUpsCoords.add(utmUps);
+      }
+    }
+    if (utmUpsCoords.isEmpty()) {
+      LOGGER.trace("No valid UTM or UPS strings could be inferred from query [{}]", query);
+      return null;
+    }
+    return new LiteralSuggestion(
+        LITERAL_SUGGESTION_ID,
+        nameBuilder.toString(),
+        utmUpsCoords
+            .stream()
+            .map(this::toLatLon)
+            .filter(Objects::nonNull)
+            .map(d -> new LatLon(d.getLat(), d.getLon()))
+            .collect(Collectors.toList()));
+  }
+
+  /**
+   * USNG's error handling API is inconsistent. Some inputs will trigger an {@link
+   * IllegalArgumentException} while some trigger a {@link java.text.ParseException}. For the
+   * purposes of this class, all errors are the same if the string could not be parsed.
+   *
+   * @param utmUps the utm or ups string to parse.
+   * @return a valid coordinate instance, or null if the parsing failed.
+   */
+  @Nullable
+  private UtmUpsCoordinate parseUtmUpsString(String utmUps) {
```

Comment:
❓ Could we return a mock UtmUpsCoordinate instead of proxying the actual translatorImpl, in case the translatorImpl throws an error?

Reply:
Good idea. Fixed.